src/pages/VenueLandingPage/VenueLandingPage.tsx

```diff
@@ -116,6 +128,56 @@ export const VenueLandingPage: React.FunctionComponent<VenueLandingPageProps> =
     }
   }, [venue]);
 
+  const onJoinClick = useCallback(() => {
+    if (venueId) {
+      const venueEntrance = venue?.entrance && venue.entrance.length;
+      window.location.href =
+        user && !venueEntrance
+          ? venueInsideUrl(venueId)
+          : venueEntranceUrl(venueId);
+    }
+  }, [user, venue, venueId]);
+
+  const renderJoinButton = useCallback(() => {
```

Comment:
Also, can we check for `venue`'s existence somewhere 'higher up' in this code, so that it is guaranteed to exist further down (or if it is already guaranteed to exist, cleanup any `?` accessors that aren't required)

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1148